test/Microsoft.IdentityModel.Tokens.Tests/IdentityComparerTests.cs

```diff
@@ -141,15 +334,327 @@ public void CompareKeyInfoTest()
         }
 
         [Fact]
-        public void CompareWsFederationConfigurationTest()
+        public void CompareOpenIdConnectConfigurations()
+        {
+            TestUtilities.WriteHeader($"{this}.CompareOpenIdConnectConfigurations", true);
+            var context = new CompareContext($"{this}.CompareOpenIdConnectConfigurations");
+            var config1 = new OpenIdConnectConfiguration { EndSessionEndpoint = Guid.NewGuid().ToString() };
+            var config2 = new OpenIdConnectConfiguration { EndSessionEndpoint = Guid.NewGuid().ToString() };
+            IdentityComparer.AreEqual(config1, config2, context);
+
+            Assert.True(context.Diffs.Count(s => s == "EndSessionEndpoint:") == 1);
+        }
+
+        [Fact]
+        public void CompareOpenIdConnectMessages()
+        {
+            TestUtilities.WriteHeader($"{this}.CompareOpenIdConnectMessages", true);
+            var context = new CompareContext($"{this}.CompareOpenIdConnectMessages");
+            var message1 = new OpenIdConnectMessage { Prompt = Guid.NewGuid().ToString() };
+            var message2 = new OpenIdConnectMessage { Prompt = Guid.NewGuid().ToString() };
+            IdentityComparer.AreEqual(message1, message2, context);
+
+            Assert.True(context.Diffs.Count(s => s == "Prompt:") == 1);
+        }
+
+        [Fact]
+        public void CompareReferences()
+        {
+            TestUtilities.WriteHeader($"{this}.CompareReferences", true);
+            var context = new CompareContext($"{this}.CompareReferences");
+            var reference1 = new Reference { Type = Guid.NewGuid().ToString() };
+            var reference2 = new Reference { Type = Guid.NewGuid().ToString() };
+            IdentityComparer.AreEqual(reference1, reference2, context);
+
+            Assert.True(context.Diffs.Count(s => s == "Type:") == 1);
+        }
+
+        [Fact]
+        public void CompareRsaSecurityKeys()
         {
-            TestUtilities.WriteHeader($"{this}.CompareKeyInfoTest", true);
-            var context = new CompareContext($"{this}.CompareKeyInfoTest");
+            TestUtilities.WriteHeader($"{this}.CompareRsaSecurityKeys", true);
+            var context = new CompareContext($"{this}.CompareRsaSecurityKeys");
+            var rsaSecurityKey1 = new RsaSecurityKey(KeyingMaterial.RsaParameters1);
+            var rsaSecurityKey2 = new RsaSecurityKey(KeyingMaterial.RsaParameters1);
+            TestUtilities.SetField(rsaSecurityKey1, "_hasPrivateKey", true);
+            TestUtilities.SetField(rsaSecurityKey2, "_hasPrivateKey", false);
+            IdentityComparer.AreEqual(rsaSecurityKey1, rsaSecurityKey2, context);
+
+            Assert.True(context.Diffs.Count(s => s == "HasPrivateKey:") == 1);
+            Assert.True(context.Diffs.Count(s => s == "Parameters:") == 1);
```

Comment:
Just created a new issue for this:
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/686